src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,385 @@
+  m_Union :=
+  begin
+    intros _ hf₁ hf₂,
+    rw [μ.m_Union hf₁ hf₂, ennreal.tsum_to_real_eq, if_pos (measurable_set.Union hf₁),
+        summable.has_sum_iff],
+    { congr, ext n, rw if_pos (hf₁ n) },
+    { refine @summable_of_nonneg_of_le _ (ennreal.to_real ∘ μ ∘ f) _ _ _ _,
+      { intro, split_ifs,
+        exacts [ennreal.to_real_nonneg, le_refl _] },
+      { intro, split_ifs,
+        exacts [le_refl _, ennreal.to_real_nonneg] },
+        exact vector_measure.summable_measure hf₁ hf₂ },
+    { intros a ha,
+      apply ne_of_lt hμ.measure_univ_lt_top,
+      rw [eq_top_iff, ← ha, outer_measure.measure_of_eq_coe, coe_to_outer_measure],
+      exact measure_mono (set.subset_univ _) }
+  end }
+
+lemma to_signed_measure_apply_measurable {μ : measure α} [finite_measure μ]
+  {i : set α} (hi : measurable_set i) :
+  μ.to_signed_measure i = (μ i).to_real :=
+if_pos hi
+
+/-- A measure is a vector measure over `ℝ≥0∞`. -/
+def to_vector_measure (μ : measure α) : vector_measure α ℝ≥0∞ :=
+{ measure_of := λ i : set α, if measurable_set i then μ i else 0,
+  empty := by simp [μ.empty],
+  not_measurable := λ _ hi, if_neg hi,
+  m_Union := λ _ hf₁ hf₂,
+  begin
+    rw summable.has_sum_iff ennreal.summable,
+    { rw [if_pos (measurable_set.Union hf₁), measure_theory.measure_Union hf₂ hf₁],
+      exact tsum_congr (λ n, if_pos (hf₁ n)) },
+  end }
+
+@[simp]
+lemma to_vector_measure_apply {μ : measure α}
+  {i : set α} (hi : measurable_set i) :
+  μ.to_vector_measure i = μ i :=
+if_pos hi
+
+end measure
+
+namespace vector_measure
+
+variables {M : Type*} [add_comm_group M] [topological_space M]
+
+/-- The zero signed measure. -/
+def zero : vector_measure α M :=
+⟨0, rfl, λ _ _, rfl, λ _ _ _, has_sum_zero⟩
+
+instance : has_zero (vector_measure α M) := ⟨zero⟩
+instance : inhabited (vector_measure α M) := ⟨0⟩
+
+@[simp]
+lemma zero_apply (i : set α) : (0 : vector_measure α M) i = 0 := rfl
+
+variables [topological_add_group M]
+
+/-- The negative of a vector measure is a vector measure. -/
+def neg (v : vector_measure α M) : vector_measure α M :=
+{ measure_of := -v,
+  empty := by simp,
+  not_measurable := λ _ hi, by simp [v.of_not_measurable_set hi],
+  m_Union := λ f hf₁ hf₂,
+    has_sum.neg $ v.has_sum_of_disjoint_Union hf₁ hf₂ }
+
+/-- The sum of two vector measure is a vector measure. -/
+def add (v w : vector_measure α M) : vector_measure α M :=
+{ measure_of := v + w,
+  empty := by simp,
+  not_measurable := λ _ hi,
+    by simp [v.of_not_measurable_set hi, w.of_not_measurable_set hi],
+  m_Union := λ f hf₁ hf₂,
+    has_sum.add (v.has_sum_of_disjoint_Union hf₁ hf₂)
+      (w.has_sum_of_disjoint_Union hf₁ hf₂) }
+
+instance : has_add (vector_measure α M) := ⟨add⟩
+instance : has_neg (vector_measure α M) := ⟨neg⟩
+
+@[simp]
+lemma coe_neg {v : vector_measure α M} (i : set α) :
+  (-v) i = - v i := rfl
+
+@[simp]
+lemma coe_add {v w : vector_measure α M} (i : set α) :
+  (v + w) i = v i + w i := rfl
```

Comment:
This lemma name corresponds to the lemma
```suggestion
  \u=(v + w) = v + w := rfl
```
But with the unicode substitution actually applied!

Reply:
Similarly for `coe_neg` and `coe_sub` I assume?